superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx

```diff
@@ -59,21 +59,23 @@ const renderOverlay = () => (
 
 export default function Header({
   setDataset,
-  datasetName,
+  title,
+  schema,
 }: {
   setDataset: React.Dispatch<DSReducerActionType>;
-  datasetName: string;
+  title: string;
+  schema?: string | null | undefined;
 }) {
   const editableTitleProps = {
-    title: datasetName,
-    placeholder: t('Add the name of the dataset'),
+    title: schema ? title : t('New dataset'),
+    placeholder: t('New dataset'),
```

Comment:
Since you are using t'(New dataset') twice here, it may be worth defining a const with that value and using it in both places.  If you export that const you could also use it in the test file so that if someone changes the value later you don't have to keep three separate string literals in sync manually.

export const DEFAULT_TITLE = t(New dataset);

Then use that on line 70, 71, and in line 51 of Header.test.tsx

Reply:
Updated in [`this commit`](https://github.com/apache/superset/pull/21557/commits/e9f6eb28b1a684b08eb097d43dee99034d775df8)